llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -5375,6 +5383,10 @@ def : Pat<(v16i8 (concat_vectors
                  (v8i8 (trunc (smax (smin (v8i16 V128:$Vn), (v8i16 VImm7F)),
                                           (v8i16 VImm80)))))),
           (SQXTNv16i8 (INSERT_SUBREG (IMPLICIT_DEF), V64:$Vd, dsub), V128:$Vn)>;
+def : Pat<(v16i8 (concat_vectors
```

Comment:
It looks like we are missing some patterns for the unsigned upper "2" cases, I will see about adding them and they can be witched over too. Hopefully the new node will be able to handle i64->i32 cases too.

Reply:
There are some extra test cases in llvm/test/CodeGen/AArch64/qmovn.ll (but only for legal cases), and I added the missing patterns in cbebacef5e12c985ca32c894bcfddec9adb2fc85.